src/stores/fileStore.ts

```diff
@@ -38,6 +47,16 @@ export class FileStore implements IFileStore {
     const msg = await telescoperApi.getMessage(fileId, messageIndexWithinFile)
     this.setCurrentMessage(msg)
   }
+
+  @action.bound
+  async getFile(fileId: string) {
+    const file = await telescoperApi.getFile(fileId)
+    if (file) {
+      this.setFile(file)
+    } else {
+      this.setFile({id: '', filename: ''})
```

Comment:
Why not set file to undefined?

Reply:
Thanks @orndorffgrant . Setting it to `undefined` was throwing a `ts` exception. It may have something to do with how we've defined the `File` interface

Reply:
Aha, typescript is complaining because the prototype of `setCurrentFile` specifies that the first argument must be an  `HL7File`, and `undefined` is not an `HL7File`.

In typescript, putting a `?` after a parameter name basically means "or undefined". So `file: HL7File` means that `file` *has* to be an `HL7File`, but `file?: HL7File` means that `file` can be an `HL7File` OR `undefined`. 

When we specify the `currentFile` field in our store, we use a `?` so it can be undefined.
So we can change the prototype of `setCurrentFile` from `setCurrentFile(file: HL7File): void;` to `setCurrentFile(file?: HL7File): void;` and everything should be okay. Then you can pass in `undefined` here as an argument.

Reply:
Totally worked! Awesome!!!